lib/document/DocumentFactory.js

```diff
@@ -97,12 +97,12 @@ class DocumentFactory {
   async createFromObject(rawDocument, options = {}) {
     const opts = { skipValidation: false, ...options };
 
-    if (!opts.skipValidation) {
-      const result = await this.fetchAndValidateDataContract(rawDocument);
+    const result = await this.fetchAndValidateDataContract(rawDocument);
 
-      if (result.isValid()) {
-        const dataContract = result.getData();
+    const dataContract = result.getData();
 
+    if (!opts.skipValidation) {
+      if (result.isValid()) {
```

Comment:
result validity we need to check right after `fetchAndValidateDataContract`. We can't create a document without his data contract.